docs/src/pages/docs/agents/01-agent-memory.mdx

```diff
@@ -7,6 +7,91 @@ description: Documentation on how agents in Mastra use memory to store conversat
 
 Agents in Mastra have a sophisticated memory system that stores conversation history and contextual information. This memory system supports both traditional message storage and vector-based semantic search, enabling agents to maintain state across interactions and retrieve relevant historical context.
```

Comment:
More comments on the API generally:
- Having both `getMessages` and `rememberMessages` is confusing. `Memory.search()` should handle both use-cases
- `addMessages` is too verbose, consider just `.add({})`, also consider whether it should also take an array of messages

Reply:
Threads should probably have `name`s rather than `title`s

Reply:
In the memory class reference, if you have a nested object inside objects, in this case `historySearch` (or `semanticRecall` or whatever you end up calling it, you should break it out into its own section with a header

https://mastra.ai/docs/reference/memory/Memory

Reply:
Example should show the simple case, `2` not `{ before: 2, after: 2 }` -- https://mastra.ai/docs/reference/memory/Memory

Reply:
The reason for `getMessages` and `rememberMessages` was due to timing with getting storage classes shipped - `getMessages` was compatible with storage while `rememberMessages` was new. Now that all the storage stuff is done combining makes sense for sure.

I think `search` will be confusing since it's more about querying (no search terms are required) - so I can combine them into `Memory.search()` if you're good with that

I can update thread title to name too - all the thread stuff is pre-existing in the public API (but somewhat undocumented).

I like `semanticSearch` a lot, I'll update that as well

Reply:
I was looking at `getMessages` and `rememberMessages` to combine them and it came back to me why these are separate.

`getMessages` is our standardized way of querying messages (this can be renamed to `query`) 
`rememberMessages` is specifically related to how the memory strategy chooses to implement memory - in our memory strategy this translates from the specific strategy options `options: { semanticSearch, lastMessages }` to the standard across all memory strategy implementation interface in `getMessages` - specifically the `selectBy` arg

Reply:
So we do need both of them but we can rename getMessages->query and either keep as-is or rename rememberMessages.

Main thing though is that getMessages will be the same across memory implementations while rememberMessages is specific to an individual implementation.

I took a crack at renaming `title` to `name` but I discarded the changes since there were way too many. I think it's out of scope for this work since it was pre-existing, but I can make a story for it as a follow up.

I did rename to `semanticSearch` though